contractcourt/channel_arbitrator.go

```diff
@@ -717,7 +717,19 @@ func (c *ChannelArbitrator) stateStep(triggerHeight uint32,
 		log.Infof("ChannelArbitrator(%v): still awaiting contract "+
 			"resolution", c.cfg.ChanPoint)
 
-		nextState = StateWaitingFullResolution
+		numUnresolved, err := c.log.FetchUnresolvedContracts()
```

Comment:
This not fixing a bug, but merely a refactoring to make the flow more understandable.

Previously, the channel arbitrator state machine could finish in `StateWaitingFullResolution`, because it was short cut when the last contract resolved. In that case, also the persistent state wasn't advanced.

Also, `c.cfg.MarkChannelResolved()` was called in two places.